settings.php

```diff
@@ -41,24 +43,59 @@
     );
     $ADMIN->add('tools', $settings);
 
-    $settings->add(new admin_setting_configcheckbox(
-        'tool_excimer/excimerenable',
-        new lang_string('excimerenable', 'tool_excimer'),
-        '', 0));
+    if ($ADMIN->fulltree) {
+        $settings->add(
+            new admin_setting_configselect(
+                'tool_excimer/excimerlogmethod',
+                get_string('excimerlogmethod', 'tool_excimer'),
+                get_string('excimerlogmethod_desc', 'tool_excimer'),
+                manager::LOGMETHOD_MANUAL,
+                [
+                    manager::LOGMETHOD_MANUAL => get_string('excimerlogmethod_manual', 'tool_excimer'),
+                    manager::LOGMETHOD_AUTO => get_string('excimerlogmethod_auto', 'tool_excimer')
+                ]
+            )
+        );
 
-    $settings->add(new admin_setting_configduration(
-        'tool_excimer/excimerexpiry_s',
-        new lang_string('excimerexpiry_s', 'tool_excimer'),
-        '', WEEKSECS));
+        // TODO make this appear only when logmethod is auto.
```

Comment:
Conditional admin settings are on the spectrum of hard so I'd not attempt this. It is also nice to be able to change settings even when they are not in use

Reply:
The TODO comment has been removed.